src/azure-cli/azure/cli/command_modules/redis/tests/latest/test_redis_scenario.py

```diff
@@ -264,14 +271,103 @@ def test_redis_cache_regenerate_key_update_tags(self, resource_group):
 
         self.cmd('az redis create -n {name} -g {rg} -l {location} --sku {sku} --vm-size {size}')
 
-        if self.is_live:
-            time.sleep(20 * 60)
 
         self.cmd('az redis regenerate-keys -n {name} -g {rg} --key-type Primary')
         self.cmd('az redis regenerate-keys -n {name} -g {rg} --key-type Secondary')
         self.cmd('az redis update -n {name} -g {rg} --set "tags.mytag=mytagval"')
         self.cmd('az redis delete -n {name} -g {rg} -y')
 
+    @ResourceGroupPreparer(name_prefix='cli_test_redis')
+    def test_redis_cache_managed_identity(self, resource_group):
+        self.kwargs = {
+            'rg': resource_group,
+            'name': self.create_random_name(prefix=name_prefix, length=24),
+            'location': location,
+            'sku': basic_sku,
+            'size': basic_size,
+            'userIdentity': "/subscriptions/0ee2a145-4d40-44f4-b764-67b40274f1ac/resourcegroups/prn-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/test",
+        }
+        self.cmd('az redis create -n {name} -g {rg} -l {location} --sku {sku} --vm-size {size} --mi-system-assigned --mi-user-assigned "{userIdentity}"', checks=[
+            self.check('identity.type', 'SystemAssigned, UserAssigned'),
+            self.check('length(identity.userAssignedIdentities)', 1)
+        ])
+
+        self.cmd('az redis identity remove -n {name} -g {rg} --mi-system-assigned --mi-user-assigned "{userIdentity}"',checks=[
+            self.check('type', 'None')
+        ])
+
+        if self.is_live:
+            time.sleep(30)
+
+        self.cmd('az redis identity assign -n {name} -g {rg} --mi-system-assigned',checks=[
+            self.check('type', 'SystemAssigned'),
+            self.check('userAssignedIdentities', None)
+        ])
+
+        if self.is_live:
+            time.sleep(30)
+
+        self.cmd('az redis identity assign -n {name} -g {rg} --mi-user-assigned "{userIdentity}"',checks=[
```

Comment:
This will remove system assigned identity because of no `--mi-system-assigned` given, is this expected?

Reply:
I think I can make it better by preserving any old identities

Reply:
I have changed the behavior.